src/sinol_make/commands/run/__init__.py

```diff
@@ -725,9 +748,25 @@ def convert_to_expected(results):
                 for group in config_expected_scores[solution]["expected"]:
                     used_groups.add(group)
         else:
-            for solution in results.keys():
-                for group in results[solution].keys():
-                    used_groups.add(group)
+            used_groups = self.get_whole_groups()
+
+            # This removes groups with not all tests run from `new_expected_scores` based on `used_groups`.
+            # If there was a solution with all groups removed, it is also removed.
```

Comment:
In fact, I also don't understand this line. Does a "solution" have "groups"?

Reply:
I changed the comments. What do you think now?